lib/invoca/utils/enumerable.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+# Invoca ::Enumerable extensions
+module Enumerable
+  def build_hash(res = {})
+    each do |x|
+      pair = block_given? ? yield(x) : x
+      res[pair.first] = pair.last if pair
+    end
+    res
+  end
+
+  class MultiSender
+
+    undef_method(*(instance_methods.map { |m| m.to_s } - %w*__id__ __send__ object_id*))
+
+    def initialize(enumerable, method)
+      @enumerable = enumerable
+      @method     = method
+    end
+
+    def method_missing(name, *args, &block)
+      @enumerable.send(@method) { |x| x.send(name, *args, &block) }
+    end
+
+  end
+
+  def *()
```

Comment:
Huh, any idea when this gets called? It's a special case vs. `method_missing` above being general. And it doesn't allow any arguments.

Update: I think this is how they're enabling chaining. If so, can you add a test of this? As in:
```
assert_equal ["OM", "HO", "OR"], ['some', 'short', 'words'].*.upcase.*.slice(1, 3)
```

Reply:
This is how the .* functionality gets added to Enumerable classes other than Array. These did a special patch for Array, I suppose, to preserve the multiplication behavior. Leaving this here could break other Enumerables multiplication behavior...unless I implement it like they did in Array. Which, now that I am typing this, seems like a good idea.

Reply:
Actually, no, that won't work of course! We are not pre-pending *() onto any of the classes, so the call to super makes no sense. Maybe we should just dump the support for Enumerables in general. I would imagine we only use for Arrays. And, we will get a runtime error (likely to be found in tests) if we are using it on other Enumerables. I'm going to kill it for now.

Reply:
Oh, weird.

I agree that it doesn't seem to make sense to preserve the existing `*` behavior only for `Array`--that seems like it would work fine, generally.

Also, I doubt we need that `:multiply` alias. I'm 99.9% sure no one knew it was there.

Reply:
I was actually making the opposite point. I only want to add the new * behavior to Array and leave all other enumerables untouched.